src/common/types/row/tuple_data_layout.cpp

```diff
@@ -110,10 +111,11 @@ void TupleDataLayout::Initialize(vector<LogicalType> types_p, Aggregates aggrega
 #endif
 
 	has_destructor = false;
-	for (auto &aggr : GetAggregates()) {
+	for (idx_t aggr_idx = 0; aggr_idx < aggregates.size(); aggr_idx++) {
+		const auto &aggr = aggregates[aggr_idx];
 		if (aggr.function.destructor) {
 			has_destructor = true;
-			break;
+			aggr_destructor_idxs.push_back(aggr_idx);
```

Comment:
Minor nit: maybe `has_destructor` can be removed now and replaced with `!aggr_destructor_idxs.empty()`?

Reply:
Good nit, thanks. Fixed